src/main/anovos/data_analyzer/association_evaluator.py

```diff
@@ -250,6 +250,8 @@ def variable_clustering(
     idf_encoded = cat_to_num_unsupervised(
         spark, idf_sample, list_of_cols=cat_cols, method_type="label_encoding"
     )
+    num_cols, cat_cols, other_cols = attributeType_segregation(idf_encoded)
+    idf_encoded = idf_encoded.select(num_cols)
     idf_imputed = imputation_MMM(spark, idf_encoded, stats_mode=stats_mode)
```

Comment:
Hi @kajanansangar , Imputation_MMM is checking if the dataframe's numerical and categorical columns have any missing data (null values), and impute them with values accordingly
- Numerical columns with missing data: use pyspark.ml.feature.Imputer to replace them with **median** values
- Categorical columns with missing data: read the measure_of_centralTendency table (stats_mode) and replace these categorical columns with **mode** values

It is an essential part for initiating VarClusHi(), because the package is calculating correlation matrix using `corr = np.corrcoef(df.values.T)`. This numpy function is null-intolerant, meaning that if input dataframe has any null values, the entries of generated correlation matrix will all be `nan`, stopping it from calculating eigenvals, eigvecs and further clustering process.

Reply:
We can make the imputation an option, but the user has to be very confident that no null values are in their dataset in order to successfully get the Anovos report. If null values appears and passed into VarClusHi(), the user see error `Eigenvalues did not converge` and report generation process stops.

Since the `imputation_MMM` function is already doing a check on null values before performing actual imputation, and if there's no null values in all columns (missing_cols=[]), it will just stop and return the original dataset directly.

Reply:
@ZhuLi99 noted and thanks for the clarification. Resolving the conversation